testnet/stacks-node/src/tests/neon_integrations.rs

```diff
@@ -1987,57 +1987,151 @@ fn atlas_integration_test() {
         channel.stop_chains_coordinator();
     });
 
-    let follower_node_thread = thread::spawn(move || {
-        // Start the attached observer
-        test_observer::spawn();
-
-        // The bootstrap node mined a few blocks and is ready, let's setup this node.
-        match follower_node_rx.recv() {
-            Ok(Signal::BootstrapNodeReady) => {
-                println!("Booting follower node...");
-            }
-            _ => panic!("Bootstrap node could nod boot. Aborting test."),
-        };
+    // Start the attached observer
+    test_observer::spawn();
```

Comment:
Did the follower_node not need to be spawned in a different thread?

Reply:
we only need to have one of the node on a separate thread.  I've been observing some flakiness between the integration tests, with the test_observer not being cleared (previously done in a separate thread, now on the main thread) after the atlas test. This refactoring in an attempt to fix this issue.